src/coreComponents/constitutive/solid/PorousSolid.hpp

```diff
@@ -81,13 +97,239 @@ class PorousSolidUpdates : public CoupledSolidUpdates< SOLID_TYPE, BiotPorosity,
+                                    real64 ( & componentMassContentIncrement )[NUM_MAX_COMPONENTS],
+                                    real64 ( & dComponentMassContent_dVolStrainIncrement )[NUM_MAX_COMPONENTS],
+                                    real64 ( & dComponentMassContent_dPressure )[NUM_MAX_COMPONENTS],
+                                    real64 ( & dComponentMassContent_dComponents )[NUM_MAX_COMPONENTS][NUM_MAX_COMPONENTS],
+                                    DiscretizationOps & stiffness,
+                                    real64 & poreVolumeConstraint,
+                                    real64 (&dPoreVolumeConstraint_dPressure ),
+                                    real64 (& dPoreVolumeConstraint_dComponents )[1][NUM_MAX_COMPONENTS] ) const
+  {
+    // Compute total stress increment and its derivative w.r.t. pressure
+    m_solidUpdate.smallStrainUpdate( k,
+                                     q,
+                                     strainIncrement,
+                                     totalStress, // first effective stress increment accumulated
+                                     stiffness );
+
+    updateBiotCoefficient( k );
+
+    real64 const biotCoefficient = m_porosityUpdate.getBiotCoefficient( k );
+    real64 const initialBiotCoefficient = biotCoefficient; // temporary
+    LvArray::tensorOps::symAddIdentity< 3 >( totalStress, -biotCoefficient * ( fluidPressureOld + deltaFluidPressure ) + initialBiotCoefficient * initialFluidPressure );
+
+    dTotalStress_dPressure[0] = -biotCoefficient;
+    dTotalStress_dPressure[1] = -biotCoefficient;
+    dTotalStress_dPressure[2] = -biotCoefficient;
+    dTotalStress_dPressure[3] = 0;
+    dTotalStress_dPressure[4] = 0;
+    dTotalStress_dPressure[5] = 0;
+
+    real64 dPorosity_dPressure;
+    real64 dPorosity_dVolStrain;
+    m_porosityUpdate.updateFromPressureAndStrain( k,
+                                                  q,
+                                                  deltaFluidPressure,
+                                                  strainIncrement,
+                                                  dPorosity_dPressure,
+                                                  dPorosity_dVolStrain );
+
+    //real64 const
+    real64 porosity = m_porosityUpdate.getPorosity( k, q );
+    real64 const porosityOld = m_porosityUpdate.getOldPorosity( k, q );
+    real64 const porosityInit = m_porosityUpdate.getInitialPorosity( k, q );
+
+    // Compute body force vector.
+    // No derivatives computed. Current assumptions:
+    // (  i) dMixtureDens_dVolStrain contribution is neglected
+    // ( ii) grains are assumed incompressible
+    // (iii) TODO add dMixtureDens_dPressure and dMixtureDens_dGlobalCompDensity
+    if( gravityAcceleration > 0.0 )
+    {
+      // Compute mixture density
+      real64 mixtureDensityNew = fluidPhaseSaturation( 0 ) * fluidPhaseMassDensity( 0 );
+      for( localIndex i = 1; i < NP; ++i )
```

Comment:
```suggestion
      for( integer i = 1; i < NP; ++i )
```
If you want, you can replace all the `localIndex` with `integer` in the phase/component loops of your PR. The code of the compositional solvers / fluid models are not entirely consistent on the use of `localIndex` vs `integer` yet, but they are almost there.

Reply:
Done. We should also change also the quadrature point index `q` from `localIndex` to `integer`. I will address that in another PR.